modules/util.py

```diff
@@ -62,6 +62,22 @@ def getBootPartition(conffile):
         return bootdevice
     return None
 
+def getPartitionRelativeToBoot(conffile, label, relativeIndex):
+    ''' Returns the partition device path when index is relative to boot partition '''
+    # First search by label
+    partdevice = getDevice(label)
+    if not partdevice:
+        match = re.match(r"(.*?)(\d+$)", getBootPartition(conffile))
```

Comment:
How about this? http://pyparted.sourcearchive.com/documentation/2.0.12-1/classparted_1_1disk_1_1Disk_d43678c0a74a3905997cecc68a8a34ea.html#d43678c0a74a3905997cecc68a8a34ea